src/arrayMethodReduce.js

```diff
@@ -4,8 +4,31 @@
  * Implement method Reduce
  */
 function applyCustomReduce() {
-  [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+  [].__proto__.reduce2 = function(...args) {
+    const callback = args[0];
+    let initialValue = args[1];
+
+    if (args.length === 2 && initialValue === undefined) {
+      initialValue = 'undefined';
+    }
```

Comment:
I am lost in your variable naming, so used other names just which correspond to the variable functional meaning.
```suggestion
    if (args.length < 2) {
      previousValue = this[0];
      startIndex = 1;
    }
```